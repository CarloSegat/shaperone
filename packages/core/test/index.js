import * as ns from '@tpluscode/rdf-ns-builders'
import namespace from '@rdfjs/namespace'
import cf from 'clownface'
import mocha from 'mocha'
import chai from 'chai'
import $rdf from 'rdf-ext'
import sinon from 'sinon'
import {form} from '../index.js'

const { describe, it, beforeEach } = mocha
const { expect } = chai

const ex = namespace('http://example.com/')

describe('@hydrofoil/shaperone/form', () => {
    let dataset
    let renderer

    beforeEach(() => {
        dataset = $rdf.dataset()
        renderer = {
            getResult: sinon.stub()
        }
    })

    it('adds rdf:type assertion', async () => {
        // given
        const shape = {
            targetClass: {
                id: ns.schema.Person
            },
            property: []
        }
        const resource = cf({ dataset, term: ex.foo })

        // when
        await form({
            shape,
            resource,
            matcher: {},
            renderer,
        })

        // then
        chai.expect(dataset.toCanonical()).to.matchSnapshot()
    })

    it('wraps graph pointer as shape instance', async () => {
        // given
        const shape = cf({ dataset: $rdf.dataset() })
            .blankNode()
            .addOut(ns.rdf.type, ns.sh.Shape)
            .addOut(ns.sh.targetClass, ns.schema.Person)
        const resource = cf({ dataset, term: ex.foo })

        // when
        await form({
            shape,
            resource,
            matcher: {},
            renderer,
        })

        // then
        chai.expect(dataset.toCanonical()).to.matchSnapshot()
    })
})
