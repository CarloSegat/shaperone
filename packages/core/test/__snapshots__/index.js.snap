// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@hydrofoil/shaperone/form adds rdf:type assertion 1`] = `
"<http://example.com/foo> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person> .
"
`;

exports[`@hydrofoil/shaperone/form wraps graph pointer as shape instance 1`] = `
"<http://example.com/foo> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person> .
"
`;
